apps/admin/backend/src/app.ts

```diff
@@ -120,6 +126,54 @@ function buildApi({
       );
     },
 
+    async setSystemSettings(input: {
+      systemSettings: string;
+    }): Promise<SetSystemSettingsResult> {
+      await logger.log(
+        LogEventId.SystemSettingsSaveInitiated,
+        assertDefined(await getUserRole()),
+        { disposition: 'na' }
+      );
+
+      const { systemSettings } = input;
+      const validatedSystemSettings = safeParseJson(
+        systemSettings,
+        SystemSettingsSchema
+      );
+      if (validatedSystemSettings.isErr()) {
+        return err({
+          type: 'parsing',
+          message: validatedSystemSettings.err()?.message,
+        });
+      }
+
+      let id;
+      try {
+        id = store.addSystemSettings(validatedSystemSettings.ok());
+        store.setCurrentSystemSettingsId(id);
+      } catch (error) {
+        const typedError = error as Error;
+        await logger.log(
+          LogEventId.SystemSettingsSaved,
+          assertDefined(await getUserRole()),
+          { disposition: 'failure', error: typedError.message }
+        );
+        return err({
+          type: 'database',
+          message: `Failed to save system settings to db. Error: ${typedError.message}`,
+        });
```

Comment:
We typically use [`Result`](https://github.com/votingworks/vxsuite/blob/25857c511eb778be3647db3c4a9a843bc1efc051/libs/basics/src/result.ts) error objects (`err(...)`) for expected/user error cases. So for an invalid file, this makes sense. But for a DB error, less so.

For that, I'd probably recommend letting the thrown error bubble up (as other handlers in this file do). (For context, these errors trigger the frontend error boundary.)

There are definitely some fuzzy areas when it comes to classifying errors. For example, for a lot of smart card errors, I've been using `err(...)`s because those are often triggered by user actions like preemptively removing a card from the card reader.